packages/components/src/components/pin-code/pin-code-utils.ts

```diff
@@ -0,0 +1,72 @@
+import type { FormState } from '../../utils/form/form-state';
+import type { Styles } from 'jss';
+import { setAttribute } from '../../utils';
+import { consoleWarn } from '../../utils/log';
+
+export const PIN_CODE_TYPES = ['number', 'password'] as const;
+export type PinCodeType = (typeof PIN_CODE_TYPES)[number];
+
+export const PIN_CODE_LENGTHS = [4 as number, 6 as number] as const;
+export type PinCodeLength = (typeof PIN_CODE_LENGTHS)[number];
+
+export type PinCodeUpdateEvent = { value: string[] };
+
+export type PinCodeState = FormState;
+
+export const getStylesWithoutSlottedSelector = (styles: Styles): Styles => {
+  return Object.fromEntries(
+    Object.entries(styles).map(([key, value]) => {
+      value = typeof value === 'object' ? getStylesWithoutSlottedSelector(value as Styles) : value;
+      return [key.replace(/::slotted\(([^,]+)\)/g, '$1'), value];
+    }, {} as Styles)
+  );
+};
+
+export const warnAboutTransformedInitialValue = (length?: number): void => {
+  const warningPrefix = '@Prop() "value" on component <p-pin-code>:';
```

Comment:
Please check if we usually use double quotes for formatting something like. Also what about the  formatting around `p-pin-code` and potential prefixing?